### ISSUE TITLE
[ENHANCEMENT] Adding session_id as argument for the user

### ISSUE THREAD

**Issue by D3vil0p3r** [paij0se/cligpt#5]
Hello, currently cligpt is not keeping the context of previous questions because `session_id` parameter is missing in the code.

A nice approach to implement could be an argument that specifies the session ID of the conversation in order that cligpt could be aware of the previous questions and answers. For example, the user can set a session ID as `cligpt --new-context` and it will write a numeric incremental or randomic session ID. Also a `cligpt --no-context` could be useful and it will go to remove the session ID in the config file.

**Comment by RagingTiger** [comment]
@D3vil0p3r This is a good idea, do you have any links to documentation on the `session_id`? I have been unable to find anything on it in the API docs ... but I did ask ChatGPT directly:

Prompt: **Can you show me an example of using `session_id` in a request to ChatGPT API in Go?**

Completion: Here is an example of using `session_id` in a request to ChatGPT API in Go:

```
package main

import (
	"fmt"
	"net/http"
)

func main() {
	sessionId := "12345"
	url := fmt.Sprintf("https://api.chatgpt.com/v1/chat?session_id=%s", sessionId)
	req, _ := http.NewRequest("POST", url, nil)
	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Println(err)
	}
	defer resp.Body.Close()
}
```

Prompt: **Can you help me understand how `session_id` will help the conversation with ChatGPT?**

Completion: The `session_id` helps ChatGPT remember the context of the conversation and is used to identify the conversation session. This helps ChatGPT to remember what topics were discussed and what the user's preferences and interests are. By using the `session_id`, ChatGPT can provide a more personalized experience for the user, enabling it to better understand the user's needs and provide more relevant responses.

**Comment by RagingTiger** [comment]
@D3vil0p3r Yeah it seems that `session_id` is the superior way to handle this *chat history* problem.

**Comment by falcucci** [comment]
looks interesting for a chat mode feature.

Me for example use this tool to ask random stuff, most of the time not related to each other still.

this will be useful if this tool goes to a chat mode approach also.

**Comment by RagingTiger** [comment]
> @D3vil0p3r Yeah it seems that `session_id` is the superior way to handle this _chat history_ problem.

So I attempted to implement this, and it seems that it is not part of the *accepted* API ... which means when I sent the request with the `session_id` it would return a 404. 

So at this point it seems like the only way to implement some kind of history is actually through *appending* previous conversations to the current prompt, before the current prompt is submitted to *ChatGPT* etc ...

Here is the actual *docs on completions*: https://beta.openai.com/docs/api-reference/completions

**Comment by paij0se** [comment]
Maybe using https://github.com/PullRequestInc/go-gpt3 ?
